common/src/main/java/com/cloudflare/access/atlassian/common/TokenVerifier.java

```diff
@@ -100,4 +84,18 @@ ClaimsVerifier validateIssuer() {
 		}
 	}
 
+	private static class JwtConsumer extends JoseJwtConsumer{
+		private final AuthenticationContext context;
+
+		public JwtConsumer(AuthenticationContext context) {
+			this.context = context;
+		}
+
+		@Override
+		protected JwsSignatureVerifier getInitializedSignatureVerifier(JwtToken jwt) {
+			JsonWebKey jwk = context.getJwk(jwt.getJwsHeader("kid").toString());
+			return JwsUtils.getSignatureVerifier(jwk);
```

Comment:
This will just throw a signature verification error if the kid can't be found even after invalidating the cache right?

Reply:
Yes, the verifier won't be initialized and the request will be blocked. :+1: